connector/datadogconnector/connector.go

```diff
@@ -67,15 +67,15 @@ func newConnector(set component.TelemetrySettings, _ component.Config, metricsCo
 		in:              in,
 		metricsConsumer: metricsConsumer,
 		tracesConsumer:  tracesConsumer,
-		exit:            make(chan struct{}),
 	}, nil
 }
 
 // Start implements the component.Component interface.
 func (c *connectorImp) Start(_ context.Context, _ component.Host) error {
 	c.logger.Info("Starting datadogconnector")
 	c.agent.Start()
-	if c.metricsConsumer != nil {
+	c.exit = make(chan struct{})
```

Comment:
Why this change? Seems no change compared to doing it in `newConnector`

Reply:
See #30487 for reasoning